test/hotspot/jtreg/serviceability/sa/UniqueVtableTest.java

```diff
@@ -170,9 +188,11 @@ private void run() throws Throwable {
     public static void main(String... args) throws Throwable {
         SATestUtils.skipIfCannotAttach(); // throws SkippedException if attach not expected to work.
 
-        UniqueVtableTest test = new UniqueVtableTest();
-
-        test.run();
+        if (args == null || args.length == 0) {
+            runMain();
+        } else {
+            runTest(Long.parseLong(args[0]));
+        }
```

Comment:
Could use some comments here. Also, I think `SATestUtils.skipIfCannotAttach` is only needed for the `else` part.

Reply:
"else" part is a sub-process.
As far as I understand it SATestUtils.skipIfCannotAttach can be skipped for "else", but it's needed for main process.

Reply:
Added comment.
Left SATestUtils.skipIfCannotAttach as is (this is consistent with other SA tests)

Reply:
> "else" part is a sub-process.

Ok, I read it backwards. Part of the reason I asked for comments. :)